### PR TITLE
New tamil cheat sheet

### PR DESCRIPTION
**Pull request by saumyasurendiran**
**What does your Instant Answer do?**
It will display all basic Tamil phrases

**What is the data source for your Instant Answer? (Provide a link if possible)**
(n/a)

**Why did you choose this data source?**
(n/a)

**Are there any other alternative (better) data sources?**
(n/a)

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Students, Tourists

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
https://duck.co/ideas/idea/6823/tamil-language-cheatsheet

**Are you having any problems? Do you need our help with anything?**
 As of now nothing i am being guided well in ddg.

**Where did you hear about DuckDuckHack? (For first time contributors)**
 From a friend(Gautham Krishna R)

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

https://files.slack.com/files-pri/T07QZ47NV-F0K18V1QV/pasted_image_at_2016_01_21_02_12_pm.png

https://duck.co/ia/view/tamil_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@saumyasurendiran Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by gautamkrishnar** [comment]
@saumyasurendiran Congratulations on your first contribution :small_blue_diamond: :smile:

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/language/tamil.json [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  "description": "Basic Tamil words and phrases",
+  "metadata": {
+    "sourceName": "Linguanaut",
+    "sourceUrl": "http://www.linguanaut.com/english_tamil.html"
+  },
+  "aliases": [
+    "Tamil phrases",
+    "english to Tamil",
+    "basic Tamil phrases",
+    "basic Tamil"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/language/tamil.json]: @saumyasurendiran Lets avoid upper case letters in aliases

File: share/goodie/cheat_sheets/json/language/tamil.json [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      },
+      {
+        "key": "Stop!",
+        "val": "நிறுத்து!",
+        "trn": "Niruthu!"
+      },
+      {
+        "key": "Do you speak English?",
+        "val": "நீங்கள் ஆங்கிலம் பேசுவீர்களா?",
+        "trn": "neengal aangilam pesuveergala "
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/language/tamil.json]: @saumyasurendiran I think here is an extra space at the end of val field.
Lets remove it

File: share/goodie/cheat_sheets/json/language/tamil.json [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        "key": "What's your name?",
+        "val": "உங்கள் பெயர் என்ன ?",
+        "trn": "ungal peyar enna?"
+      }
+    ],
+    "Travelling": [
+      {
+        "key": "Turn left.",
+        "val": "இடப்பக்கம் திரும்பு .",
+        "trn": "idappakkam thirumbu."
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/language/tamil.json]: @saumyasurendiran Lets remove periods from end of phrases

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

**Comment by gautamkrishnar** [comment]
@jophab thanks for finding :+1: 
@moollaza good to review now...

File: share/goodie/cheat_sheets/json/language/tamil.json [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        "key": "How can I get a taxi?",
+        "val": "நான் ஒரு டாக்ஸி பெற முடியும்?",
+        "trn": "Naan oru taxi pera mudiyuma?"
+      }
+    ],
+    "Going out for dinner": [
+      {
+        "key": "A cup of coffee please.",
+        "val": "ஒரு கப் காபி தயவு செய்து கொடுக்க.",
+        "trn": "oru cup coffee thaiyavu seidhu kuduka."
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/language/tamil.json]: @saumyasurendiran Is kuduka same as  "kudunga"?
Or do you have a typo here?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/language/tamil.json (hunk 4 above)
Comment (saumyasurendiran) [inline share/goodie/cheat_sheets/json/language/tamil.json]: thats a typo i changed

Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/language/tamil.json]: :+1:

File: share/goodie/cheat_sheets/json/language/tamil.json [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+      {
+        "key": "Can you repeat, please?",
+        "val": "தயவு செய்து மீண்டும் கூற முடியுமா?",
+        "trn": "thaiyavu seidhu meendum kura mudiyuma"
+      },
+      {
+        "key": "How do you say something in Tamil?",
+        "val": "நீங்கள் எப்படி தமிழ் ஏதாவது சொல்ல?",
+        "trn": "neengal tamil lil epadi edavadu solla"
+      },
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/language/tamil.json]: @saumyasurendiran  I think you missed a question mark end end of trn field for above three fields.
Lets add those.
some of following also have to be corrected from the same.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/language/tamil.json [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+      {
+        "key": "What's your name?",
+        "val": "உங்கள் பெயர் என்ன ?",
+        "trn": "ungal peyar enna?"
+      }
+    ],
+    "Travelling": [
+      {
+        "key": "Turn left.",
+        "val": "இடப்பக்கம் திரும்பு .",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/language/tamil.json]: @saumyasurendiran Lets remove the periods too.
You can do it easily by finding all ." (dot followed by double quotes) and replace it with " (doublequotes)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/language/tamil.json [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+      {
+        "key": "Can you repeat, please?",
+        "val": "தயவு செய்து மீண்டும் கூற முடியுமா?",
+        "trn": "thaiyavu seidhu meendum kura mudiyuma?"
+      },
+      {
+        "key": "How do you say something in Tamil?",
+        "val": "நீங்கள் எப்படி தமிழ் ஏதாவது சொல்ல?",
+        "trn": "neengal tamil lil epadi edavadu solla?"
+      },
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/language/tamil.json]: @saumyasurendiran Sorry about not mentioning it earlier.
make the capitalisation of first letter of val field consistent

File: share/goodie/cheat_sheets/json/language/tamil.json [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+      {
+        "key": "Stop!",
+        "val": "நிறுத்து!",
+        "trn": "Niruthu!"
+      },
+      {
+        "key": "Do you speak English?",
+        "val": "நீங்கள் ஆங்கிலம் பேசுவீர்களா?",
+        "trn": "neengal aangilam pesuveergala"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/language/tamil.json]: @saumyasurendiran missing question mark at trn field.
lets go through entire phrases and add necessary !, ?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/language/tamil.json [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+        "trn": "engae?"
+      },
+      {
+        "key": "Is it far?",
+        "val": "அது தூரத்தில் இருக்கிறது ?",
+        "trn": "Athu dhoorathil irukiradha?"
+      },
+      {
+        "key": "Please stop here.",
+        "val": "இங்கே நிறுத்துங்கள் .",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/language/tamil.json]: @saumyasurendiran Also go through entire phrases and remove periods.

File: share/goodie/cheat_sheets/json/language/tamil.json [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+    ],
+    "Going out for dinner": [
+      {
+        "key": "A cup of coffee please",
+        "val": "ஒரு கப் காபி தயவு செய்து கொடுக்க",
+        "trn": "oru cup coffee thaiyavu seidhu kudunga"
+      },
+      {
+        "key": "Where is a good restaurant?",
+        "val": "எங்கே ஒரு நல்ல உணவகம் உள்ளது??",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/language/tamil.json]: @saumyasurendiran an extra ? here

Comment (saumyasurendiran) [inline share/goodie/cheat_sheets/json/language/tamil.json]: have taken an extra ?

**Comment by jophab** [comment]
@saumyasurendiran Lets make capitalization of first letter in val filed consistent.
Also go through entire phrases and add necessary !,?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

**Comment by saumyasurendiran** [comment]
yeah have modified them in capital ones

File: share/goodie/cheat_sheets/json/language/tamil.json [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+      },
+      {
+        "key": "Stop!",
+        "val": "நிறுத்து!",
+        "trn": "Niruthu!"
+      },
+      {
+        "key": "Do you speak English?",
+        "val": "நீங்கள் ஆங்கிலம் பேசுவீர்களா?",
+        "trn": "Neengal aangilam pesuveergala"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/language/tamil.json]: @saumyasurendiran just this one remaining

**Comment by jophab** [comment]
@saumyasurendiran This LGTM :+1: 
Thanks for your quick responses. :+1: 
wait for @moollaza

**Comment by saumyasurendiran** [comment]
yup sure its my pleasure :)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@saumyasurendiran @jophab I take it you're native Tamil speakers?

Typically we tweet out a call for feedback if we don't have any native speakers. But if you're both sure other Tamil speakers will approve of this then I think its good to merge

**Comment by saumyasurendiran** [comment]
i speak tamil:)

Sent from my ASUS

-------- Original Message --------

**Comment by gautamkrishnar** [comment]
@moollaza yup. saumya lives in tamil nadu.

**Comment by moollaza** [comment]
Excellent :+1:

File: share/goodie/cheat_sheets/json/language/tamil.json [hunk 12]
```diff
@@ -0,0 +1,205 @@
+{
+  "id": "Tamil_cheat_sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/language/tamil.json]: whoops, this needs to be lowercased.

Comment (saumyasurendiran) [inline share/goodie/cheat_sheets/json/language/tamil.json]: yeah done

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=tamil cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/tamil_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@saumyasurendiran this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by moollaza** [comment]
@saumyasurendiran thanks again for creating the Tamil Cheat Sheet! You did a great job. We at DuckDuckGo, the DuckDuckHack community, and all the people who will encounter this Instant Answer really appreciate it.

We could really use your help on something else now. We want to increase our Instant Answer coverage for as many programming languages and platforms as possible. We want an OpenGL Cheat Sheet and I think you'd similarly do a great job with building that. Please update the [Programming IA Coverage table](https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage) (mark it with a 'd') if you'd be willing to take it up!

I think it is a good fit for you, but I'm of course not entirely sure, so I'd love to chat if you're around this week or next. Please message me on slack (@moollaza) or send me an email (moollaza@duckduckgo.com) if you'd be interested in talking. 

Thanks again!

**Pull request merged**